src/editor/editor_display.cpp

```diff
@@ -139,4 +159,44 @@ display::overlay_map& editor_display::get_overlays()
 	return controller_.get_current_map_context().get_overlays();
 }
 
+void editor_display::clear_help_string()
+{
+	font::remove_floating_label(help_handle_);
+	help_handle_ = 0;
+}
+
+void editor_display::set_help_string(const std::string& str)
+{
+	clear_help_string();
+
+	const color_t color{0, 0, 0, 0xbb};
+
+	int size = font::SIZE_LARGE;
+	point canvas_size = video::game_canvas_size();
+
+	while(size > 0) {
+		if(font::pango_line_width(str, size) * 2 > canvas_size.x) {
+			size--;
+		} else {
+			break;
+		}
+	}
+
+	const int border = 5;
+
+	font::floating_label flabel(str);
+	flabel.set_font_size(size);
+	flabel.set_position(canvas_size.x / 2, canvas_size.y);
```

Comment:
The comment about possible cleanup was prompted by this line, which could avoid the need to call `font::move_floating_label` later in this function.